diff --git a/src/plugins/contrib/wxSmith/wxwidgets/properties/wxsstyleproperty.cpp b/src/plugins/contrib/wxSmith/wxwidgets/properties/wxsstyleproperty.cpp
index 47861d7b7..aada3a7f9 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/properties/wxsstyleproperty.cpp
+++ b/src/plugins/contrib/wxSmith/wxwidgets/properties/wxsstyleproperty.cpp
@@ -25,6 +25,7 @@
 #include "../wxsflags.h"
 
 #include <wx/tokenzr.h>
+#include <wx/propgrid/propgrid.h>
 
 // Helper macro for fetching variables
 #define STYLEBITS   wxsVARIABLE(Object,Offset,long)
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/properties/wxscolourproperty.cpp b/src/plugins/contrib/wxSmith/wxwidgets/properties/wxscolourproperty.cpp
index 1474aab1c..6e4a75264 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/properties/wxscolourproperty.cpp
+++ b/src/plugins/contrib/wxSmith/wxwidgets/properties/wxscolourproperty.cpp
@@ -480,7 +480,11 @@ namespace
         wxASSERT( propgrid );

         // Must only occur when user triggers event
+#if wxCHECK_VERSION(3, 3, 0)
+        if ( !(propgrid->GetInternalFlags() & wxPropertyGrid::wxPG_FL_IN_HANDLECUSTOMEDITOREVENT) )
+#else
         if ( !(propgrid->GetInternalFlags() & wxPG_FL_IN_HANDLECUSTOMEDITOREVENT) )
+#endif
             return res;

         wxColourPropertyValue val = GetVal();
@@ -578,7 +582,11 @@ namespace
             int index = paintdata.m_choiceItem;
             value = wxsColourValues[index];
         }
+#if wxCHECK_VERSION(3, 3, 0)
+        else if ( !(m_flags & wxPGPropertyFlags(wxPG_EX_WINDOW_STYLES::wxPG_PROP_UNSPECIFIED)) )
+#else
         else if ( !(m_flags & wxPG_PROP_UNSPECIFIED) )
+#endif
         {
             value = GetVal().m_type;
         }
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/wxsitemeditorcontent.h b/src/plugins/contrib/wxSmith/wxwidgets/wxsitemeditorcontent.h
index 57b42059a..78442594f 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/wxsitemeditorcontent.h
+++ b/src/plugins/contrib/wxSmith/wxwidgets/wxsitemeditorcontent.h
@@ -23,6 +23,7 @@
 #ifndef WXSITEMEDITORCONTENT_H
 #define WXSITEMEDITORCONTENT_H

+#include <wx/hashmap.h>
 #include "wxsdrawingwindow.h"

 class wxsItemEditorDragAssist;

---
