From 396009e0012687d482124420068cf16d586c7696 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Fri, 16 Aug 2019 09:26:19 -0400
Subject: Edit winpr/include/winpr/nt.h

---
 winpr/include/winpr/nt.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/winpr/include/winpr/nt.h b/winpr/include/winpr/nt.h
index d7ba79590..0293a46cc 100644
--- a/winpr/include/winpr/nt.h
+++ b/winpr/include/winpr/nt.h
@@ -1281,15 +1281,22 @@ static INLINE NTSTATUS NTSTATUS_FROM_WIN32(long x)
 #define FILE_INFORMATION_CLASS		_WINTERNL_FILE_INFORMATION_CLASS
 #define _FILE_INFORMATION_CLASS		_WINTERNL__FILE_INFORMATION_CLASS
 #define FileDirectoryInformation	_WINTERNL_FileDirectoryInformation
+#define FILE_DIRECTORY_INFORMATION	_WINTERNL_FILE_DIRECTORY_INFORMATION
+#define _FILE_DIRECTORY_INFORMATION	_WINTERNL__FILE_DIRECTORY_INFORMATION
 
 #include <winternl.h>
 
+#ifndef __MINGW64_VERSION_MAJOR
 #undef FILE_INFORMATION_CLASS
 #undef _FILE_INFORMATION_CLASS
 #undef FileDirectoryInformation
+#endif
+#undef FILE_DIRECTORY_INFORMATION
+#undef _FILE_DIRECTORY_INFORMATION
 
 #endif
 
+#ifndef __MINGW64_VERSION_MAJOR
 typedef enum _FILE_INFORMATION_CLASS
 {
 	FileDirectoryInformation = 1,
@@ -1333,6 +1340,7 @@ typedef enum _FILE_INFORMATION_CLASS
 	FileValidDataLengthInformation,
 	FileShortNameInformation
 } FILE_INFORMATION_CLASS;
+#endif
 
 #if !defined(_WIN32) || defined(_UWP)
 
@@ -1375,8 +1383,11 @@ typedef enum _FILE_INFORMATION_CLASS
 #define FILE_VALID_PIPE_OPTION_FLAGS		0x00000032
 #define FILE_VALID_MAILSLOT_OPTION_FLAGS	0x00000032
 #define FILE_VALID_SET_FLAGS			0x00000036
+#endif
 
 #define FILE_SUPERSEDED				0x00000000
+
+#if !defined(_WIN32) || defined(_UWP)
 #define FILE_OPENED				0x00000001
 #define FILE_CREATED				0x00000002
 #define FILE_OVERWRITTEN			0x00000003
-- 
