From d6b2c08249a5162ede5904880364671db3a6fc3d Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Fri, 16 Aug 2019 09:52:15 -0400
Subject: Edit winpr/include/winpr/smartcard.h

---
 winpr/include/winpr/smartcard.h | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/winpr/include/winpr/smartcard.h b/winpr/include/winpr/smartcard.h
index a372ae3ad..779374a5b 100644
--- a/winpr/include/winpr/smartcard.h
+++ b/winpr/include/winpr/smartcard.h
@@ -83,10 +83,15 @@
 #define SCARD_E_COMM_DATA_LOST			(HRESULT)(0x8010002FL)
 #define SCARD_E_NO_KEY_CONTAINER		(HRESULT)(0x80100030L)
 #define SCARD_E_SERVER_TOO_BUSY			(HRESULT)(0x80100031L)
+#endif
+
+#ifndef SCARD_E_PIN_CACHE_EXPIRED
 #define SCARD_E_PIN_CACHE_EXPIRED		(HRESULT)(0x80100032L)
 #define SCARD_E_NO_PIN_CACHE			(HRESULT)(0x80100033L)
 #define SCARD_E_READ_ONLY_CARD			(HRESULT)(0x80100034L)
+#endif
 
+#ifndef SCARD_W_UNSUPPORTED_CARD
 #define SCARD_W_UNSUPPORTED_CARD		(HRESULT)(0x80100065L)
 #define SCARD_W_UNRESPONSIVE_CARD		(HRESULT)(0x80100066L)
 #define SCARD_W_UNPOWERED_CARD			(HRESULT)(0x80100067L)
@@ -100,8 +105,10 @@
 #define SCARD_W_CARD_NOT_AUTHENTICATED		(HRESULT)(0x8010006FL)
 #define SCARD_W_CACHE_ITEM_NOT_FOUND		(HRESULT)(0x80100070L)
 #define SCARD_W_CACHE_ITEM_STALE		(HRESULT)(0x80100071L)
-#define SCARD_W_CACHE_ITEM_TOO_BIG		(HRESULT)(0x80100072L)
-
+#endif
+
+#ifndef SCARD_W_CACHE_ITEM_TOO_BIG
+#define SCARD_W_CACHE_ITEM_TOO_BIG		((DWORD)0x80100072L)
 #endif
 
 #define SCARD_ATR_LENGTH			33
-- 
