From fa291fd10fd3cf01f2c12c41c2ff2dd2a7027535 Mon Sep 17 00:00:00 2001
From: LIU Hao <lh_mouse@126.com>
Date: Wed, 22 Mar 2023 16:14:58 +0800
Subject: [PATCH] gcc: Make stupid AT&T syntax not default

Signed-off-by: LIU Hao <lh_mouse@126.com>
---
 gcc/config/i386/i386.opt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/i386/i386.opt b/gcc/config/i386/i386.opt
index 94fdd639ff1..56319fbf8e4 100644
--- a/gcc/config/i386/i386.opt
+++ b/gcc/config/i386/i386.opt
@@ -263,7 +263,7 @@ Target RejectNegative Negative(march=) Joined Var(ix86_arch_string)
 Generate code for given CPU.
 
 masm=
-Target RejectNegative Joined Enum(asm_dialect) Var(ix86_asm_dialect) Init(ASM_ATT)
+Target RejectNegative Joined Enum(asm_dialect) Var(ix86_asm_dialect) Init(ASM_INTEL)
 Use given assembler dialect.
 
 Enum
-- 
2.34.1

